redshift_sqlalchemy/dialect.py

```diff
@@ -19,6 +23,53 @@ class RedshiftImpl(postgresql.PostgresqlImpl):
         __dialect__ = 'redshift'
 
 
+## Constants
+
+COMPOUND_SORTKEY = 'compound'
+INTERLEAVED_SORTKEY = 'interleaved'
+
+
+## Regular expressions
+
+
+# For parsing adsrc, e.g.: "identity"(445178, 0, '1,1'::text)
+IDENTITY_RE = re.compile(r"""
+    "identity" \(
+      (?P<current>-?\d+)
+      ,\s
+      (?P<base>-?\d+)
+      ,\s
+      '(?P<seed>-?\d+),(?P<step>-?\d+)'
+      .*
+    \)
+""", re.VERBOSE)
+
+# For constraints, e.g.:
+#   FOREIGN KEY(col1) REFERENCES othertable (col2)
+# See https://docs.aws.amazon.com/redshift/latest/dg/r_names.html
+# for a definition of valid SQL identifiers.
+FK_RE = re.compile(r"""
+  ^FOREIGN KEY \s* \(    # FOREIGN KEY, arbitrary whitespace, literal '('
```

Comment:
Either should be valid as long as we're not in multiline mode. `^` is more familiar.